src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -447,6 +447,16 @@ private String findRubyHome() throws IOException {
                 final File jar = new File(codeSource.getLocation().getFile());
                 final File jarDir = jar.getParentFile();
 
+                final File jarDirParent = jarDir.getParentFile();
+                final File jarDirGrandparent = jarDirParent == null ? null : jarDirParent.getParentFile();
+
+                if (jarDir.getName().equals("dists")
+                        && jarDirParent != null && jarDirParent.getName().equals("mxbuild")
+                        && jarDirGrandparent != null && new File(jarDirGrandparent, "lib").isDirectory()) {
+                    // Source build
```

Comment:
Is that a graalvm source build or a normal TruffleRuby source build?
Why is it needed, which launcher does not set -Xhome?